### DISCUSSION TITLE
montage images with preserve orignal size

### DISCUSSION THREAD

**Discussion by ldurniat** [ImageMagick/ImageMagick#4844] [Help] [answered]
Hi,

I want create tileset from single images without shrink or enlarge them. Unfortunately IM enlarge images to fit into 448 x 448 tile.

Command:

`magick montage -geometry 448x448 block.png buttonHelp.png buttonStart.png cloud.png helpBackground.png title.png images.png`

Name/size

block.png 64x64
buttonHelp.png 192x64
buttonStart.png 192x64
cloud.png 384x128
helpBackground.png 416x416
title.png 448x224

Output:
![images](https://user-images.githubusercontent.com/22639656/154137732-bf9e59a6-823f-4f97-add5-7b923e6a1cf8.png)

Using 448x448> in command above remove block.png from output and make it 0KB.

I use Windows 10 and IM  ImageMagick 7.1.0-24 Q16-HDRI x64 2022-02-12

Have a nice day:)

**Comment by fmw42** [comment]
If you do not want the resize, then leave off the -geometry or use -geometry +X+Y to put X and Y spacing between them. If you want all image to fill some size, then you should pad them first and then feed that to montage.

`magick block.png buttonHelp.png buttonStart.png cloud.png helpBackground.png title.png -gravity center -background white -extent 448x448 miff:- | magick montage -  images.png`

I suggest you may want to specify -tile NxM and -geometry +X+Y to specify the tiling arrangement and the spacing between the tiles.  If you want no spacing, then use -geometry +0+0. The default is +2+2. I do not know the default for -tile.

see https://legacy.imagemagick.org/Usage/montage/#geometry_size

**Reply by ldurniat** [reply]
Your command result in image of 384x252. It is smaller than the largest image used to create tileset (output).

Output:
![images](https://user-images.githubusercontent.com/22639656/154142217-9c4c167e-bfe6-483b-a39f-2545413d14a2.png)

**Reply by fmw42** [reply]
You said you did not want the image resized.  So please clarify exactly what you want for the output!

**Reply by ldurniat** [reply]
> You said you did not want the image resized. So please clarify exactly what you want for the output!

I expected: 

block.png 64x64 -> 448x448
buttonHelp.png 192x64 -> 448x448
buttonStart.png 192x64 -> 448x448 Image is bigger but button itself have the same size and centered on tile.
cloud.png 384x128 -> 448x448
helpBackground.png 416x416 -> 448x448
title.png 448x224 -> 448x448
So output image size should be 1344x896 for tileset 3x2 (as above).

**Comment by snibgo** [comment]
I think the OP doesn't want "-extent 448x448".

**Reply by fmw42** [reply]
Then as I said in my answer, the first option was just removing -geometry or using -geometry +X+Y in his command.

**Comment by fmw42** [comment, marked as answer]
Then  add ! to your geometry to force resize to exactly that size and so try

`magick montage  block.png buttonHelp.png buttonStart.png cloud.png helpBackground.png title.png -geometry "448x448!+0+0" -tile 3x2 images.png`

If that does not work, then post each of your input images separately so that we can test with them. You can zip them together and post the zip file if you want.

**Reply by ldurniat** [reply]
> `magick montage block.png buttonHelp.png buttonStart.png cloud.png helpBackground.png title.png -geometry "448x448!+0+0" -tile 3x2 images.png`

I use "448x448>+0+0" instead of "448x448!+0+0". Before I don't use quotes so it don't work properly.

Full comand
`magick montage block.png buttonHelp.png buttonStart.png cloud.png helpBackground.png title.png  -geometry "448x448>+0+0" -tile 3x2 images.png`

Output:
![images](https://user-images.githubusercontent.com/22639656/154283419-9d84959b-bdf4-422e-8e31-bec6ebb24f9b.png)


Thanks for help:)
Have a nice day:)
